analysis/mod.sankeymaker.php

```diff
@@ -129,6 +129,10 @@
 
 
         <?php
+        $dataset = htmlentities($dataset);
```

Comment:
This line is necessary and baffled me until I found this bit of code in [analysis.common.functions.php](https://github.com/digitalmethodsinitiative/dmi-tcat/blob/11625b8f277ccf085161b2b3a6db7efcb5e03cb9/analysis/common/functions.php#L13). It looks like this will always pick up query parameters for analysis. I think we could apply htmlentities or htmlspecialchars to all the $_GET calls there.

The [validate_all_variables](https://github.com/digitalmethodsinitiative/dmi-tcat/blob/11625b8f277ccf085161b2b3a6db7efcb5e03cb9/analysis/common/functions.php#L799) function ought to be catching anything that could go into the database, but it does still allow someone to pull @stijn-uva's little `?dataset="><script src="data:text/javascript;charset=utf-8;base64,YWxlcnQoJ2hhY2tlZC4uLiBieSB0aGUgaWxsdW1pbmF0aScp"type="text/javascript" language="javascript"></script><input name="fake" type="hidden" value="` trick.